lux/core/frame.py

```diff
@@ -242,6 +244,26 @@ def set_intent_as_vis(self, vis: Vis):
         self._intent = vis._inferred_intent
         self._parse_validate_compile_intent()
 
+    def asLuxType(self, types: dict):
```

Comment:
Do we have a way to check if users have specified valid datatypes? Wondering if there should be some quick validation to see if all the types set by a user are either nominal, quantitative, id, or temporal.

Reply:
Ahhh, great catch. I am now throwing an error if it is not in those 4. How does that sound?

Reply:
Yup the new changes look good!